code-challenges.test.js

```diff
@@ -1,59 +1,194 @@
+describe ("upperCaseArray", () => {
+                it("returns array with capital letters", () => {
+                     expect(upperCaseArray(["streetlamp", "potato", "teeth", "conclusion", "nephew"])).toEqual(["Streetlamp", "Potato", "Teeth", "Conclusion", "Nephew"]),
+                     expect(upperCaseArray(["temperature", "database", "chopsticks", "mango"])).toEqual(["Temperature", "Database", "Chopsticks", "Mango"])
+        })
+    })
+//     ReferenceError: upperCaseArray is not defined
+
+//     117 | describe ("upperCaseArray", () => {
+//     118 |                 it("returns array with capital letters", () => {
+//   > 119 |                      expect(upperCaseArray(["streetlamp", "potato", "teeth", "conclusion", "nephew"])).toEqual(["Streetlamp", "Potato", "Teeth", "Conclusion", "Nephew"]),
+//         |                      ^
+//     120 |                      expect(upperCaseArray(["temperature", "database", "chopsticks", "mango"])).toEqual(["Temperature", "Database", "Chopsticks", "Mango"])
+//     121 |         })
+//     122 |     })
+
+//     at Object.expect (code-challenges.test.js:119:22)
+
 
 // b) Create the function that makes the test pass.
 
+const upperCaseArray = (array) => {
+    return array.map((value) => {
+        return value[0].toUpperCase() + value.substring(1)
+       
+    })
+}
+console.log(upperCaseArray(randomNouns1))
+console.log(upperCaseArray(randomNouns2))
+
+// upperCaseArray
+//  Output:   ✓ returns array with capital letters
 // --------------------3) Create a function that takes in a string and logs the index of the first vowel.
-
+// Function: Declare an function vowelFinder that takes in a string and outputs the index of the first vowel in the string. 
+// Input: A string of letters
+// Output: Index of the first vowel in the string given.
 // a) Create a test with expect statements for each of the variables provided.
-
+describe ("vowelFinder", () => {
+    it("returns the index of the first vowel", () => {
+        expect(vowelFinder("learn")).toEqual(1)
+        expect(vowelFinder("academy")).toEqual(0)
+        expect(vowelFinder("challenges")).toEqual(2)
+    })
+})
 const vowelTester1 = "learn"
 // Expected output: 1
 const vowelTester2 = "academy"
 // Expected output: 0
 const vowelTester3 = "challenges"
 // Expected output: 2
 
+// ReferenceError: vowelFinder is not defined
+
+//       151 | describe ("vowelFinder", () => {
+//       152 |     it("returns the index of the first vowel", () => {
+//     > 153 |         expect(vowelFinder("learn")).toEqual(1)
+//           |         ^
+//       154 |         expect(vowelFinder("academy")).toEqual(0)
+//       155 |         expect(vowelFinder("challenges")).toEqual(2)
+//       156 |     })
+
+//       at Object.expect (code-challenges.test.js:153:9)
+
+
 // b) Create the function that makes the test pass.
+const vowelList = /[a,e,i,o,u]/i;
+const vowelFinder = (string) => {
+    return string.match(vowelList).index
```

Comment:
So .match is a method that finds the result that is a match to what parameters (or stipulations) you are requesting. In this case, you made a list of vowels (vowelList) and you then made a function that will take in a string, and return if there is a match between the given string and the list of vowels. Let me know if this helped or you would like me to explain it anymore (:

Reply:
It did help thanks Nicole!! My apologies for the late reply.